client/my-sites/earn/main.jsx

```diff
@@ -101,6 +102,48 @@ class EarningsMain extends Component {
 		this.props.dismissWordAdsError( siteId );
 	};
 
+	/**
+	 * Remove any query parameters from the path before using it to
+	 * identify which screen the user is seeing.
+	 *
+	 * @returns {String} Path to current screen.
+	 */
+	getCurrentPath = () => {
+		let currentPath = this.props.path;
+		const queryStartPosition = currentPath.indexOf( '?' );
+		if ( queryStartPosition > -1 ) {
+			currentPath = currentPath.substring( 0, queryStartPosition );
+		}
+		return currentPath;
+	};
+
+	/**
+	 * Check the current path and returns an appropriate title.
+	 *
+	 * @returns {String} Header text for current screen.
+	 */
+	getHeaderText = () => {
+		const { translate } = this.props;
+
+		switch ( this.getCurrentPath().replace( /^\/earn\/([-a-z]*)(\/.*)?$/gm, '$1' ) ) {
+			case 'payments':
+				return translate( 'Recurring Payments' );
+
+			case 'ads-earnings':
+				return translate( 'Ads Earnings' );
+
+			case 'ads-settings':
+				return translate( 'Ads Settings' );
+		}
```

Comment:
Do we need a default here? Is there a chance we'd return nothing, and if so, should we redirect somewhere?

Reply:
Thanks. I'll add a default just to be defensive, although if we have nothing at this level, something is very broken upwards and should be fixed there.